### DISCUSSION TITLE
Feedback v2.0 - Finally the Apple Watch is detected!

### DISCUSSION THREAD

**Discussion by corgan2222** [ESPresense/ESPresense#79] [General]
The latest Update kills it! Now the Apple Watch looks different as the iPhone and reports even the name correct.
Looks pretty good!

![grafik](https://user-images.githubusercontent.com/12233951/136041433-75f85da0-2bf1-49ad-a48d-ab762ef15908.png)

- Are these extreme long room Assistant name intended? Is this what I have to enter the HA Config? 
- One thing I mentioned is, that the room assistant detection came really late. I would say after around 2-3 minutes.
- At first, I had weird error/debugs messages. But they are gone now after a few minutes.

> ld_pdu_get_tx_flush_nb HCI packet count mismatch (1, 2)
> E NimBLEClient: "Connection failed; status=574 "
> E NimBLEClient: "Connection failed; status=13 "
> lld_pdu_get_tx_flush_nb HCI packet count mismatch (1, 2)
> E NimBLEClient: "Connection failed; status=574 "
> E NimBLERemoteService: "characteristicDiscCB() rc=7 "
> E NimBLERemoteService: "Could not retrieve characteristics"
> E NimBLEClient: "Disconnected, could not retrieve services -aborting"
> E NimBLEClient: "Disconnected, could not retrieve services -aborting"

great Job, thanks!!

**Comment by DTTerastar** [comment]
1. Yes, the long room assistant name is intended, that is the id to put into the config
2. Yeah, something is still a bit wonky, working on making it more consistent
3. Yeah, those are the bowls of ESP32 saying it didn't understand where to route packets.  We need a bunch of retries to work around it atm

**Reply by DTTerastar** [reply]
I put in an issue for 2: https://github.com/h2zero/NimBLE-Arduino/issues/299

**Comment by corgan2222** [comment]
Nice! Do you have also this problem, or is this because of the other sensors?

**Reply by DTTerastar** [reply]
Yes, overnight I saw my watch respond to the query about 75% of the time, which means there were a BUNCH of 15 minute segments where the watch was listed as "not_home" because it was using the non room assistant fingerprint.  I plan on using #81 to make the reliability of query much less important since we'll be able to "carry over" the query when the mac address rolls.

**Reply by DTTerastar** [reply]
It's def a beta feature... That's why I made it opt-in.

**Comment by corgan2222** [comment]
Understand. I will monitor the behavior overnight with three devices.

**Reply by DTTerastar** [reply]
The other issue is some of the ESPs might be too far to successfully transmit the query and receive it.  The normal advertisements are basically just one way so we can read stuff that is really far away.  ATM if your "far" base station detects it Queries it, and fails after 10 attempts it won't notice if the device comes closer and fix the pid based on the query.  I think the fix for this will be to "gossip" on a mqtt channel and pass the query info from the near base station to all the other ones so they can all have the extra info.

**Comment by DTTerastar** [comment]
Try v2.0.1.  The bugs I was seeing was because of multiple advertisements w/ the same mac.  You had a 50% chance of getting the right fingerprint when using roomAssistant app.

**Comment by corgan2222** [comment]
Nice, thanks!!
I'm going to Amsterdam in some hours till Thursday on a little holiday, so I don't have the time atm. 
But will check asap when I'm back.
